Modix.Data/Models/Core/MessageEntity.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Utilities;
+
+namespace Modix.Data.Models.Core
+{
+    public class MessageEntity
+    {
+        [Required]
+        public ulong Id { get; set; }
+
+        [Required]
+        public ulong GuildId { get; set; }
+
+        [Required]
+        public ulong ChannelId { get; set; }
+
+        [Required]
+        public ulong AuthorId { get; set; }
+
+        [Required]
+        public DateTimeOffset Timestamp { get; set; }
+
```

Comment:
What happened to the idea of storing a message hash, so we could filter out duplicates?

Reply:
That seemed a little controversial so I left it this time.